### ISSUE TITLE
License

### ISSUE THREAD

**Issue by Loki-Astari** [BartMassey/printf-tests#1]
Hi Bart,

I have used your tests in one of my GitHub projects.

https://github.com/Loki-Astari/ThorsIOUtil

The tests are in this file:
https://github.com/Loki-Astari/ThorsIOUtil/blob/master/src/ThorsIOUtil/test/testlist.h

Currently I use the MIT License for this project.
Additionally in the `testlist.h` file I point the reader back to your original project.

Just wanted to see if you were OK with this or if you would like some additional alterations.

Loki

**Comment by BartMassey** [comment]
Thanks much for your interest in `printf-tests`!

*[Disclaimer: I am not an attorney. Nothing in this comment is reliable legal advice.]*

tl;dr: Do what you like, as I don't really care to pursue any legal action at this point, but the legal road you're on is full of rocks.

Although I don't really care enough to do anything about it right now, this looks to me like a GPL violation as shipped. One could argue [*Feist*](https://en.wikipedia.org/wiki/Feist_Publications,_Inc.,_v._Rural_Telephone_Service_Co.) if the comments were stripped out altogether, I guess, although collection copyright probably applies here since this is not a comprehensive collection of `printf` tests. So nevermind.

If you were to write an `awk` script which generated C code to run the tests out of `printf-tests.txt`, you could probably legally distribute the tests along with that, on the argument that the tests weren't "an integral part" of the shipped product and the generated code and the generator were your own.

Another approach would be to just keep the tests internal and not ship them. The GPL allows this. You could ship your infrastructure and point people at my repo to get the tests.

Easiest would be to just license your thing GPL v2 or later. :-) As of right now, I'm not seeing any license info in your repo anyhow…

GL with your project!
